### PR TITLE
feat: collapse sidebar groups

### PR DESCRIPTION
**Pull request by juliajforesti**
[SIDE-178](https://rocketchat.atlassian.net/browse/SIDE-178)

## Proposed changes (including videos or screenshots)

- add collapsible groups to sidebar 
- save to localStorage collapsed groups - keeps state after reload
Kapture 2024-10-18 at 12 12 11

## Further comments

[SIDE-178]: https://rocketchat.atlassian.net/browse/SIDE-178?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/meteor/client/sidebarv2/RoomList/RoomList.tsx [hunk 1]
```diff
@@ -66,14 +44,27 @@ const RoomList = () => {
...
 
 	return (
 		<Box position='relative' display='flex' overflow='hidden' height='full' flexGrow={1} flexShrink={1} flexBasis='auto' ref={ref}>
 			<GroupedVirtuoso
-				groupCounts={groupCounts}
-				groupContent={(index) => <SidebarV2GroupTitle title={t(groupList[index])} />}
-				itemContent={(index) => <RoomListRow data={itemData} item={roomList[index]} />}
+				groupCounts={groupsCount}
+				groupContent={(index) => (
+					<SidebarV2CollapseGroup
```
Comment (aleksandernsilva) [inline apps/meteor/client/sidebarv2/RoomList/RoomList.tsx:58]: Refreshing the page with a group open seem to leave a gap after the section, even when the group is collapsed. It fixes itself if you collapse the group and refresh again.

Screenshot 2024-10-18 at 12 02 50

Comment (aleksandernsilva) [inline apps/meteor/client/sidebarv2/RoomList/RoomList.tsx:58]: I also noticed that the collapse arrow sometimes don't animate for me. 
_(In the gif I'm clicking once in each section)_

sidebar-colapse-gif

**Review by aleksandernsilva — CHANGES_REQUESTED** [review]

File: apps/meteor/client/sidebarv2/RoomList/RoomList.tsx (hunk 1 above)
Comment (juliajforesti) [reply apps/meteor/client/sidebarv2/RoomList/RoomList.tsx:58]: did you run `yarn && yarn build`? it seems to me you're not running with the correct fuselage version

Comment (juliajforesti) [reply apps/meteor/client/sidebarv2/RoomList/RoomList.tsx:58]: here too: `yarn && yarn build` should fix that

Comment (aleksandernsilva) [reply apps/meteor/client/sidebarv2/RoomList/RoomList.tsx:58]: Yep, that solved it 🚀

**Review by aleksandernsilva — APPROVED** [review]

**Review by ggazzo — APPROVED** [review]

**Review by aleksandernsilva — APPROVED** [review]

**Pull request merged**